src/main/java/ay2122s1_cs2103t_w16_2/btbb/storage/JsonAdaptedOrder.java

```diff
@@ -72,6 +105,56 @@ public Order toModelType() throws IllegalValueException {
         }
         final Address modelClientAddress = new Address(clientAddress);
 
-        return new Order(modelClientName, modelClientPhone, modelClientAddress);
+        if (recipeName == null) {
+            throw new IllegalValueException(String.format(
+                    MISSING_FIELD_MESSAGE_FORMAT, GenericString.getMessageConstraints("Recipe Name")
+            ));
+        }
+        if (!GenericString.isValidGenericString(recipeName)) {
+            throw new IllegalValueException(GenericString.getMessageConstraints("Recipe Name"));
+        }
+        final GenericString modelRecipeName = new GenericString(recipeName);
+
+        final List<Ingredient> ingredients = new ArrayList<>();
+        for (JsonAdaptedIngredient ingredient : recipeIngredients) {
+            Ingredient i = ingredient.toModelType();
+            boolean hasValidQuantity = GenericQuantity.isValidQuantity(i.getQuantity().toString());
```

Comment:
Is this supposed to use isValidInternalQuantity, since its converting from Json to Model
```suggestion
            boolean hasValidQuantity = GenericQuantity.isValidInternalQuantity(i.getQuantity().toString());
```

Reply:
This is correct. For all ingredients in the recipe list, the quantity is > 0 or else it is not valid.